docs/upgrade-guides/core/3-20-to-3-21.mdx

```diff
@@ -336,3 +336,235 @@ The `change_user_address` method has been removed from the Plugin Manager and is
 In previous versions, orders that had any unapproved fulfillments and no returns, were given status `PARTIALLY_FULFILLED`. This happened even when no items were actually fulfilled. 
 Currently, fulfillments with `WAITING_FOR_APPROVAL` status are excluded from determination of order status. Adding unapproved fulfillment to order will no longer result in a status change.
 Order can only receive `PARTIALLY_FULFILLED` status if there exists `FULFILLED` fulfillment and there are some items that are either unfulfilled or awaiting approval. 
+
+
+## Changes to discount propagation in orders created from checkout
+
```

Comment:
I would add a very short intro, that summarizes these changes at a high level (like I'm five years old). Example:

In Saleor 3.21 we have unified the way of how the discount propagation works for orders. _Here a quick explanation how it worked before and how it works now._ Below you'll find the details how discount propagation works for different discount types.